src/app/register/abstract-register-user.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Router } from '@angular/router';
+import { UserService } from '../services/user.service';
+import { ConfigService } from '../services/config.service';
+import { Directive, OnInit } from '@angular/core';
+import { GoogleUser } from '../modules/google-sign-in/GoogleUser';
+
+@Directive()
+export abstract class AbstractRegisterUserComponent implements OnInit {
+  protected googleAuthenticationEnabled: boolean = false;
+  protected microsoftAuthenticationEnabled: boolean;
+  protected abstract joinFormPath: string;
```

Comment:
This can be ordered alphabetically.

```microsoftAuthenticationEnabled``` can be initialized to false to be consistent with ```googleAuthenticationEnabled``` being initialized to false.

Reply:
Something we can discuss as a team: everything would continue to work even when microsoftAuthenticationEnabled and googleAuthenticationEnabled aren't initialized (to false, in this case). It would default to ```undefined```, which is ```falsy```. We do this in many parts of our code, so it'd be good to establish a convention.